applications/dashboard/controllers/class.entrycontroller.php

```diff
@@ -503,30 +503,40 @@ public function connect($method) {
         }
 
         $isTrustedProvider = $this->data('Trusted');
+        $saveRoles = false;
+        $saveRolesRegister = false;
 
         // Check if we need to sync roles
-        if (($isTrustedProvider || c('Garden.SSO.SyncRoles')) && $this->Form->getFormValue('Roles', null) !== null) {
-            $saveRoles = $saveRolesRegister = true;
-
-            // Translate the role names to IDs.
+        if (($isTrustedProvider || c('Garden.SSO.SyncRoles'))) {
             $roles = $this->Form->getFormValue('Roles', null);
-            $roles = RoleModel::getByName($roles);
-            $roleIDs = array_keys($roles);
 
-            // Ensure user has at least one role.
-            if (empty($roleIDs)) {
-                $roleIDs = $this->UserModel->newUserRoleIDs();
+            // Allow the roles index to be lower case.
+            // This feature was requested and approved here https://github.com/vanillaforums/estimates/issues/886
+            $loweredRoles = $this->Form->getFormValue('roles', null);
```

Comment:
We can probably use   
$roles = $this->Form->getFormValue('Roles',  $this->Form->getFormValue('roles', null));

Reply:
I think you are right, I did this over-complicated logic to prevent from affecting any existing plugin. However, I don't think the suggestion you made will affect anything.